src/main/python/ayab/signal_sender.py

```diff
@@ -28,26 +37,76 @@ class SignalSender(object):
     @author Tom Price
     @date   July 2020
     """
-    def __init__(self, signal_receiver, *args, **kwargs):
+    def __init__(self, signal_receiver:SignalReceiver, *args:Any, **kwargs:Any)->None:
         self.__signal_receiver = signal_receiver
-        for s in self.__signal_receiver.signals():
-            setattr(self, "emit_" + s, self.__make_emit_method(s))
         super().__init__(*args, **kwargs)
 
-    def __make_emit_method(self, signal):
-        return lambda *args: getattr(self.__signal_receiver, signal).emit(*args)
+    def emit_start_row_updater(self,start_row:int)->None:
+        self.__signal_receiver.start_row_updater.emit(start_row)
+    
```

Comment:
breaking all of these out makes the typing _much_ clearer. Which should help avoid issues at invocation sites.